### PR TITLE
Add Mega-Linter in list of integrating tools

### PR DESCRIPTION
**Pull request by nvuillam**
Hi, thanks for your linter :) 

I integrated it in Mega-Linter, can I put a link here ? :)

Internal documentation is https://nvuillam.github.io/mega-linter/descriptors/scala_scalafix/, do you have remarks ?

Best regards :)

### CODE REVIEW

**Comment by bjaglin** [comment]
Thanks for the integration @nvuillam.

One gotcha is that - as far as I can see - you are running `scalafix` without prior scalax compilation & semanticdb generation, so you cannot provide any of the [semantic options](https://nvuillam.github.io/mega-linter/descriptors/scala_scalafix/#help-content) to the CLI. That means that usage of scalafix is limited to [built-in syntactic rules](https://scalacenter.github.io/scalafix/docs/rules/overview.html). I think it would be useful to mention this limitation in your documentation and in the link on the Scalafix documentation.

**Comment by nvuillam** [comment]
Thanks for your reply :)

Is there a way to generically perform scalax compilation and semanticdb generation ?

- If yes I may be able to run them before calling scalafix
- If not I can just update the doc to specify the use of scalafix in Mega-Linter is limited to built-in syntactic rules

**Comment by bjaglin** [comment]
> Is there a way to generically perform scalax compilation and semanticdb generation via cli ?

scalafix may compile and generate sources on demand, but this feature is not documented and very fragile, so the short answer is no. That's why I also suggest to simply mention this limitation rather than trying to overcome it.

**Comment by nvuillam** [comment]
Updated  !

Is is ok like that ? https://nvuillam.github.io/mega-linter/descriptors/scala_scalafix/

**Comment by nvuillam** [comment]
I also updated my PR (online in github editor, so you may use Squash & merge option to avoid useless commits in master )

File: docs/users/installation.md [hunk 1]
```diff
@@ -446,6 +446,7 @@ Scalafix is supported in other build tools via externally maintained plugins:
 - Mill: [mill-scalafix](https://github.com/joan38/mill-scalafix)
 - Gradle: [gradle-scalafix](https://github.com/cosmicsilence/gradle-scalafix)
 - Maven: [scalafix-maven-plugin](https://github.com/evis/scalafix-maven-plugin)
+- [Mega-Linter](https://nvuillam.github.io/mega-linter/): More than 70 linters ready to use for GitHub Actions, other CI tools and locally, including [scalafix](https://nvuillam.github.io/mega-linter/descriptors/scala_scalafix/) activated by default (limited to [built-in syntactic rules](https://scalacenter.github.io/scalafix/docs/rules/overview.html), as there is no prior compilation)
```
Comment (bjaglin) [inline docs/users/installation.md:449]: I don't think it's worth mentioning the other linters on this page
```suggestion
- [Mega-Linter](https://nvuillam.github.io/mega-linter/descriptors/scala_scalafix) (only built-in syntactic rules are supported)
```

**Review by bjaglin — COMMENTED** [review]
A suggestion for the intro of https://nvuillam.github.io/mega-linter/descriptors/scala_scalafix/ focusing on the user-facing limitations

Scalafix lints and fixes scala files
* Semantic rules are not supported
* [External rules](https://scalacenter.github.io/scalafix/docs/rules/external-rules.html) are not supported

**Comment by nvuillam** [comment]
as you wish :)

**Pull request merged**